File: fintechplatform/src/main/java/com/fintechplatform/android/iban/db/IbanPersistanceDBModule.java
package com.fintechplatform.android.iban.db;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class IbanPersistanceDBModule {

    @Provides
    @Singleton
    IbanPersistanceDB providesIbanPersistance(IbanDB ibanDB) {
        return new IbanPersistanceDB(ibanDB);
    }

    @Provides
    @Singleton
    IbanDB providesIbanDB() {
        return new IbanDB();
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/iban/db/IbanPersistanceDBModule.kt
package com.fintechplatform.ui.iban.db

import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class IbanPersistanceDBModule {

    @Provides
    @Singleton
    internal fun providesIbanPersistance(ibanDB: IbanDB): IbanPersistanceDB {
        return IbanPersistanceDB(ibanDB)
    }

    @Provides
    @Singleton
    internal fun providesIbanDB(): IbanDB {
        return IbanDB()
    }
}